papas/datatypes/Event.h

```diff
@@ -58,7 +58,7 @@ namespace papas {
 class Event {
 public:
   /// @brief  Constructor
-  Event(std::shared_ptr<Nodes> hist = std::make_shared<Nodes>(Nodes()));
+  Event(Nodes& hist);
```

Comment:
Sorry, I don't understand your question. Is this what you are after:-
typedef DAG::Node<Identifier> PFNode;
typedef std::map<Identifier, PFNode> Nodes;